homeassistant/components/zwave_js/cover.py

```diff
@@ -116,6 +192,117 @@ async def async_stop_cover(self, **kwargs: Any) -> None:
             await self.info.node.async_set_value(target_value, False)
 
 
+class FGR222Venetian(ZWaveCover):
```

Comment:
We probably don't want to add device specific code to the platforms. That should be solved some other way, preferably upstream in the zwave-js project.

Reply:
`zwave_js` implements the Z-Wave protocol. AFAICS (please correct me if I'm wrong), the `Manufacturer Proprietary` command class of the Z-Wave protocol is explicitly designed to contain device-specializations. By providing access to the `Manufacturer Proprietary` values, `zwave_js` is doing its part by implementing the transport layer. It mustn't know about the payload.

It's up to the driver to make use of the payload, and the driver is home-assistant. We had the same approach when we implemented this for OpenZWave (https://github.com/home-assistant/core/pull/29701) and it was merged.

Reply:
We'll only accept this PR if the device specific code can be abstracted into a problem case in discovery like we did here for dynamic climate temperature values: https://github.com/home-assistant/core/pull/49804

Reply:
Understand the ask, but I won't have time to re-design this. So feel free to close this if no-one wants to step in.

Two thoughts before you do that:
First, in the spirit of pragmatism, why introduce an abstraction for a problem now, before it is clear if there ever will be a second user of said abstraction (there might be, not denying that). The PR in its current shape will introduce less code than with the abstraction. Why not defer it to the time a second user emerges? The PR as it is right now will probably enable a bunch of users of the FGR222 right away (https://community.home-assistant.io/t/fibaro-fgrm222-venetian-blind-tilt-position/50710/17)

Second, each device that has some manufacturer-proprietary characteristics might have some unique quirks here or there. For the FGR222, for example, it is the tilt-position interfering with the reported cover position (https://github.com/home-assistant/core/pull/49371/files#diff-e6292e73b3e851acd39ec60f47355d032083c995521448faa206c8105c620fe6R252-R255). Of course, you can abstract over this specific problem as well, but doing that for each new device can bloat the abstraction over time as well.  I still think that viewing Z-Wave as a transport / communication means and not as the driver itself makes sense as well.

Reply:
We don't want device specific code in the platforms. I don't think we'll accept the workaround for the tilt. That should be fixed by the device.